sdk/tools/azure-openrewrite/src/main/java/com/azure/openrewrite/core/http/HttpHeadersCustomRecipe.java

```diff
@@ -0,0 +1,198 @@
+                // after: io.clientcore.core.http.models.HttpHeaders HttpHeader remove(io.clientcore.core.http.models.HttpHeaderName)
+                methodMatcher = new MethodMatcher("com.azure.core.http.HttpHeaders remove(java.lang.String)");
+                if (methodMatcher.matches(method, false)) {
+                    replacementTemplate = templateBuilder.getJavaTemplateBuilder("remove(HttpHeaderName.fromString(#{any(java.lang.String)}))")
+                        .imports("io.clientcore.core.http.models.HttpHeaderName")
+                        .build();
+                    method = replacementTemplate.apply(updateCursor(method), method.getCoordinates().replaceMethod(), method.getArguments().toArray());
+                    maybeAddImport("io.clientcore.core.http.models.HttpHeaderName");
+                }
+
+                // before: com.azure.core.http.HttpHeaders HttpHeader remove(com.azure.core.http.HttpHeaderName)
+                // after: no change
+
+                // before: com.azure.core.http.HttpHeaders String getValue(java.lang.String)
+                // after: io.clientcore.core.http.models.HttpHeaders String getValue(io.clientcore.core.http.models.HttpHeaderName)
+                methodMatcher = new MethodMatcher("com.azure.core.http.HttpHeaders getValue(java.lang.String)");
+                if (methodMatcher.matches(method, false)) {
+                    replacementTemplate = templateBuilder.getJavaTemplateBuilder("getValue(HttpHeaderName.fromString(#{any(java.lang.String)}))")
+                        .imports("io.clientcore.core.http.models.HttpHeaderName")
+                        .build();
+                    method = replacementTemplate.apply(updateCursor(method), method.getCoordinates().replaceMethod(), method.getArguments().toArray());
+                    maybeAddImport("io.clientcore.core.http.models.HttpHeaderName");
+                }
+
+                // before: com.azure.core.http.HttpHeaders String getValue(com.azure.core.http.HttpHeaderName)
+                // after: no change
+
+                // before: com.azure.core.http.HttpHeaders String[] getValues(java.lang.String)
+                // after: io.clientcore.core.http.models.HttpHeaders java.util.List<String> getValues(io.clientcore.core.http.models.HttpHeaderName)
+                methodMatcher = new MethodMatcher("com.azure.core.http.HttpHeaders getValues(java.lang.String)");
+                if (methodMatcher.matches(method, false)) {
+                    replacementTemplate = templateBuilder.getJavaTemplateBuilder("getValues(HttpHeaderName.fromString(#{any(java.lang.String)}))")
+                        .imports("io.clientcore.core.http.models.HttpHeaderName")
+                        .build();
+                    method = replacementTemplate.apply(updateCursor(method), method.getCoordinates().replaceMethod(), method.getArguments().toArray());
+                    maybeAddImport("io.clientcore.core.http.models.HttpHeaderName");
```

Comment:
why do we need to do both `.imports("io.clientcore.core.http.models.HttpHeaderName")` and `maybeAddImport("io.clientcore.core.http.models.HttpHeaderName");` ?

Reply:
@samvaity .`import()` tells the template builder to include the class when looking for the code that matches the template while `maybeAddImport()` tells OpenRewrite to add an import statement into the users code for the type if one doesn't  already exist.

Without `import()`, the template might not match the class type since it's not a built in java type.
Without `maybeAddImport()` the recipe might produce code that doesn't compile.